dnsimple/dns_analytics.go

```diff
@@ -0,0 +1,83 @@
+package dnsimple
+
+import (
+	"context"
+	"fmt"
+)
+
+// DnsAnalyticsService handles communication with the DNS Analytics related
+// methods of the DNSimple API.
+//
+// See https://developer.dnsimple.com/v2/dns_analytics/
+type DnsAnalyticsService struct {
+	client *Client
+}
+
+// DnsAnalytics represents DNS Analytics data.
+type DnsAnalytics struct {
+	Volume int64
+	Zone   string
+	Date   string
+}
+
+type DnsAnalyticsQueryParameters struct {
+	AccountId int64  `json:"account_id"`
+	StartDate string `json:"start_date"`
+	EndDate   string `json:"end_date"`
+	Sort      string `json:"sort"`
+	Page      int64  `json:"page"`
+	PerPage   int64  `json:"per_page"`
+	Groupings string `json:"groupings"`
+}
+
+// DnsAnalyticsResponse represents a response from an API method that returns DnsAnalytics data.
+type DnsAnalyticsResponse struct {
+	Response
+	Data    []DnsAnalytics
+	Rows    [][]interface{}             `json:"data"`
+	Headers []string                    `json:"headers"`
+	Query   DnsAnalyticsQueryParameters `json:"query"`
+}
+
+func (r *DnsAnalyticsResponse) marshallData() {
```

Comment:
`marshall` is also a valid spelling variant of `marshal`, but one less common. Not a blocker.

Reply:
I checked the existing code and we're using the single `l` spelling, so I've changed this one for consistency as you have suggested 👍